### ISSUE TITLE
Nested objects locales not working

### ISSUE THREAD

**Issue by Meierschlumpf** [QuiiBz/next-international#65]
It's not possible to access the translations by key with translate function.

It appears to be a problem caused by not using FlattenLocale.

Should look like that:
![image](https://github.com/QuiiBz/next-international/assets/63781622/cbbc83c1-4712-4157-bd3f-54791424e835)

But is currently that:
![image](https://github.com/QuiiBz/next-international/assets/63781622/850b0ce3-13af-430b-8d5b-164023679042)

**Comment by Meierschlumpf** [comment]
I guess we need to add the FlattenLocale type here:

https://github.com/QuiiBz/next-international/blob/ccc8269cb0acf3cf2946ae06d90a4bf6ae6d42dd/packages/next-international/src/app/server/index.ts#L12-L13

So it would look something like that:
```ts
  type Locale = OtherLocales extends ExplicitLocales ? GetLocaleType<OtherLocales> : FlattenLocale<GetLocaleType<Locales>>;
  type LocalesKeys = OtherLocales extends ExplicitLocales ? keyof OtherLocales : keyof FlattenLocale<Locales>;
```

**Comment by QuiiBz** [comment]
You're absolutely right. I just released a new [0.6.1 version](https://github.com/QuiiBz/next-international/releases/tag/0.6.1) that includes the fix: #66

**Comment by Meierschlumpf** [comment]
Okay thanks bro
